src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -100,12 +102,14 @@ private static Expression BuildArgumentThatMatchesAnything<TValue>()
 
         private static FakeManager GetFakeManagerCallIsMadeOn(ParsedCallExpression parsedCallExpression)
         {
-            if (parsedCallExpression.CallTarget == null)
-            {
-                throw new ArgumentException("The specified call is not made on a fake object.");
-            }
+            return parsedCallExpression.CallTarget == null
+                ? null

```

Comment:
I generally like the shape of it, but `Fake.GetFakeManager` will ultimately call `FakeFacade.GetFakeManager`, which will throw an `ArgumentNullException` when the call is static. The current behaviour is to indicate that the method is static. Unfortunately, that error is coming from the `CastleDynamicProxyInterceptionValidator`.
The existing design puts as much of the decision of whether a call can be configured into the hands of the proxy, by returning a null `FakeManager` when one can't be found. As you hint at, this is a bit of a problem, because we could then fail when we dereference the object. (Currently we won't, because there's no way have `AssertThatMemberCanBeIntercepted` pass when we can't find a `FakeManager`.)

In theory if we ever had a proxy that could support static method calls (e.g. some CodeCop proxy), we'd want to preserve this behaviour. It would mean that the way we'd find a FakeManager  would change: perhaps by looking up by target object, as we do now, for instance methods, but looking up by type if the method were static.

In practice, we're unlikely ever to have such a situation, so perhaps acting like we'll always have a faked object is reasonable.

That still leaves us with the problem of how to structure the code so it gives users the best experience and is still understandable. I think the current user experience is good, but as you point out, we're still not giving a good reading experience—why should I expect a passing `AssertThatMemberCanBeIntercepted` to mean that there's a `FakeManager` to hang rules on?

I'm wondering if we shouldn't keep all the code as is, except to explicitly check the manager for nullness before we add a rule to it. An alternative is to embrace our lack of static (sealed, non-virtual) method handling and hoist some of the checks out of the proxy interception validators, but we'd have to be careful to put them some place where they'd be hit if we tried to abuse an unnatural fake.